### ISSUE TITLE
[Bug] Alerts currently don't work if you call them from the constructor

### ISSUE THREAD

**Issue by PureWeen** [dotnet/maui#1514]
### Description

If you call DisplayAlert from the constructor of a page it currently does not work on WinUI

Go to the alerts page and add a call to DisplayAlert from the ctor

```C#
public AlertsPage()
{
      DisplayFromCtor();
}

async void DisplayFromCtor() =>  await DisplayAlert();
```

**Comment by hartez** [comment]
In Forms calling alerts from the constructor would put them in a queue that would be processed after Appearing; maybe the queue stuff didn't get ported?

**Comment by jamesmontemagno** [comment]
IMHO this should not be a supported scenario that could be documented. The UI isn't visible yet, you basically shouldn't do anything in your constructor.

**Comment by Eilon** [comment]
I'm guessing people do it because it's the logical equivalent of `printf` debugging. No strong opinion on whether it should be allowed.

**Comment by PureWeen** [comment]
Another use case here is that the ctor is easiest place for an alert you only want to show on the first load of a page. In theory you could check for this case in "Appearing" but it's a bit more code and since the ctor works then why not just put it there.

**Comment by shimingsg** [comment]
Not reproducible with Android 11.

**Comment by Eilon** [comment]
Re-opening because I'm not sure if this is the right behavior on _all_ platforms, even if perhaps it works on Android.

**Comment by jsuarezruiz** [comment]
Testing with RC3 and main branch, this is now working:
![image](https://user-images.githubusercontent.com/6755973/168038957-fe1c15d7-98a8-43fd-a1e7-942f60cb6c42.png)

Reviewing the history have been fixed by the commit fffaae1eaa33db6eeced93f0a85fc00cd155be19.
